### ISSUE TITLE
Issues with DCEG 2.5.8 (encoding of `featureName`)

### ISSUE THREAD

**Issue by DavidGrant-NIWC** [iho-ohi/S-101-Documentation-and-FC#123]
The following is currently valid and will result in a label being shown on the chart only when the national language context parameter is set to 'fra' (see highlighted bullet 2):

|name|language|name usage|
|:---|:---|:---:|
|foo|eng|3|
|bar|fra||

The 5th bullet highlighted below indicates that this encoding should result in nothing being shown on the chart. However, the 4th bullet indicates that `nameUsage`=1 is only required to indicate non-english defaults. How do you indicate that "foo" should be the default? Recommend that the 5th bullet is removed so that `nameUsage`=3 is required as the only way to limit a name to the pick report.
|name|language|name usage|
|:---|:---|:---:|
|foo|eng|
|bar|fra|

![image](https://github.com/iho-ohi/S-101-Documentation-and-FC/assets/62906211/8d7a71ed-bf04-491f-bd36-98413ae045a3)

Recommend that to simplify this modeling:
* Remove `nameUsage`=3 (no chart display). If `nameUsage` is not present then the corresponding entry will never be shown on the chart.
* Require `nameUsage`=1 (default name display) be set on the default name. This would normally be the english name, but can be set at the encoders discretion.
* Require `nameUsage`=2 (alternate name display) on each national language replacement for the default name display. You can't have an entry with `nameUsage`=2 unless there is also an entry with `nameUsage`=1.
* The default name is displayed anytime there is no match.

|name|language|name usage|
|:---|:---|:---:|
|foo|eng|1|
|long description of foo|eng|
|bar|fra|2|
|long description of bar|fra|
|blah|kor|2|

|National language|Text shown on chart|
|:---:|:---:|
|eng|foo|
|fra|bar|
|kor|blah|
|spa|foo|

**Comment by KlasOstergren-SMA** [comment]
I agree with Dave's recommendations. I think they make it a little easier to understand how to properly encode the data. Regarding the language of the default name, in most countries the default name will not be in English as most places only have names in the local language(s). For example, in Sweden there is only one place name in English (Gothenburg has both a Swedish and an English name). All other names will be in Swedish with nameUsage=1.

To make the intended coding even easier to understand, I would suggest that language=eng must be populated for all names that are in English. The only use case I can see where language should be empty is when the name is in a language that is not in ISO 639-2/T. What else happens if you encode a name in English (without populating language) and also encode an instance of a language that is not in ISO 639-2/T? I think it is better to encode language consistently when the name is in ISO 639-2/T.
This change would not lead to much more work. In our case, we could for example set up a validation check that tests that all featureName have laguage=swe and nameUsage=1. We could also create a rule that sets these attributes on all instances of featureName.

**Comment by JeffWootton** [comment]
Based on the discussion above, I have amended the first part of clause 2.5.8 as follows (track-changes are changes from DCEG Edition 1.3.0 draft):

[Clause 2.5.8 Revised_Geographic Names.docx](https://github.com/iho-ohi/S-101-Documentation-and-FC/files/15338546/Clause.2.5.8.Revised_Geographic.Names.docx)

To be discussed at S-101PT13.

**Comment by benhazelgrove** [comment]
Where Data Producers discretion comes into it, AHO see a need for the conversion mapping of Production Software to take S-57 OBJNAM attributes and have that set to 1 (default name display) in every instance. Where NOBJNM is also populated, this can be set to 2 (alternate name display) or left blank for the Producers discretion....

Having to change all named features in an ENC after conversion to be displayed seems like too much effort.

**Comment by JeffWootton** [comment]
@benhazelgrove : This will be taken into account for the next Version of S-65 Annex B.

**Comment by JeffWootton** [comment]
Modelling and encoding guidance for population of complex attribute featureName agreed at S-101PT13 and implemented in S-101 Edition 1.4.0.

Close issue. New issue(s) may be opened based on implementation and testing.
